pdpserver/control.go

```diff
@@ -164,3 +164,10 @@ func (s *server) Apply(ctx context.Context, in *pb.Update) (*pb.Response, error)
 
 	return res, err
 }
+
+func (s *server) NotifyReady(ctx context.Context, m *pb.Empty) (*pb.Response, error) {
+	log.Info("Got notified about readiness")
+	s.start <- true
```

Comment:
Instead of stopping there and then resuming here it's better to make serveRequests function and just call it here.

Reply:
Actually it is not so easy, because this function will require a tracer parameter to be passed (as the code bellow needs it). But from NotifyReady handler we have no access to it. Such approach will require some (probably not so optimal) refactoring. Global variable or additional field of the 'server' structure. I think, it is not a good idea.

Reply:
You're right. Tracer the same as interface and protocol should be landed to server struct. Issue with current code that second call to NotifyReady if happens accidentally hangs forever. There is non-blocking channel write to deal with it:
```go
select {
default:
case s.start <- true:
}
```
But I'd prefer to change s.start to s.tracer and call serveRequests here instead to be more explicit.